client/my-sites/themes/theme-showcase.jsx

```diff
@@ -207,7 +213,7 @@ class ThemeShowcase extends React.Component {
 		let callback = () => null;
 		// In this state: tabFilter = [ Recommended | ##All(1)## ]  tier = [ All(2) | Free | ##Premium## ]
 		// Clicking "Recommended" forces tier to be "all", since Recommend themes cannot filter on tier.
-		if ( 'recommended' === tabFilter && 'all' !== this.props.tier ) {
+		if ( 'recommended' === tabFilter.key && 'all' !== this.props.tier ) {
```

Comment:
The logic here and on line 200 should be expanded from 'Is on recommended' to 'Is on recommended or on trending', or perhaps more generally, 'Is on any tab except for all themes'.